ansible-scylla-node/tasks/Debian.yml

```diff
@@ -83,41 +83,24 @@
   - name: Figure out exact Scylla version
     block:
     - name: Get versions of {{ scylla_package_prefix }} package
-      shell: apt list -a {{ scylla_package_prefix }} | grep {{ scylla_version }}
+      shell: apt list -a {{ scylla_package_prefix }} | egrep "{{ scylla_version }}[\-]?"
       register: aptversions
 
     - name: Search for version of {{ scylla_package_prefix }} package
       set_fact:
         tempsversion: "{{ aptversions.stdout | regex_findall(regexp) }}"
       vars:
-          regexp: ' ([0-9\.]+)-([a-f0-9\.]+)-([0-9]) '
+        regexp: ' ([0-9\.]+)-([a-f0-9\.]+)-([0-9]) '
 
     - name: Set {{ scylla_package_prefix }} version if only one version found
       set_fact:
         sversion: "{{ tempsversion }}"
       when: tempsversion | length == 1
 
-    - name: Requery for {{ scylla_package_prefix }} version if multiple versions found
-      block:
-        - name: query for version with - at the end of {{ scylla_version }}
-          shell: apt list -a {{ scylla_package_prefix }} | grep {{ scylla_version }}-
-          register: aptversions
-
-        - name: Search for version of {{ scylla_package_prefix }} package with - postfix
-          set_fact:
-            rtempsversion: "{{ aptversions.stdout | regex_findall(regexp) }}"
-          vars:
-              regexp: ' ([0-9\.]+)-([a-f0-9\.]+)-([0-9]) '
-
-        - name: Set scylla version if only one version with - found
-          set_fact:
-            sversion: "{{ rtempsversion }}"
-          when: rtempsversion | length == 1
-
-        - name: Error out, wrong Scylla version was passed, please fix it!
-          ansible.builtin.command: /bin/false
-          when: rtempsversion | length > 1
+    - name: Error out, wrong Scylla version was passed, please fix it!
```

Comment:
You can specify why it is wrong - in this case because there is more than a single choice. Also, you don't seem to handle a case when `tempsversion | length == 0`.

Reply:
I believe the easiest check is "rtempsversion | length != 1"

Reply:
will fix

Reply:
You forgot to address my "what's wrong" remark.
You can say something like "Error out, wrong Scylla version was passed (more than a single or none version matches), please fix it!"

Reply:
fixed the error msg too now